Add BrowserButton component

cezaraugusto opened the pull request:
- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).

Auditors: @luixxiul, @NejcZdovc 
Close: #8568

> Note: This PR removed "whiteButton" for new buttons since at the point we want to allow dark-UI/theming, it will not necessarily be white.

Test Plan:

* No breaking tests should be found regarding buttons
* Buttons visible in below images should do their expected actions
* Buttons visible in below images should not have UI regressions
* about:styles should have primaryColor, secondaryColor (previous whiteButton) and wideButton aesthetically ok.

## Buttons refactored

* All buttons inside Sync tab

<img width="467" alt="screen shot 2017-04-28 at 9 29 08 pm" src="https://cloud.githubusercontent.com/assets/4672033/25552427/f6085540-2c6f-11e7-9456-84ff4af9d27b.png">
<img width="100" alt="screen shot 2017-04-28 at 9 28 18 pm" src="https://cloud.githubusercontent.com/assets/4672033/25552429/f60b5a06-2c6f-11e7-9f67-610ddad695c9.png">
<img width="725" alt="screen shot 2017-04-28 at 9 19 00 pm" src="https://cloud.githubusercontent.com/assets/4672033/25552428/f608f45a-2c6f-11e7-93ca-771a144558c0.png">
<img width="744" alt="screen shot 2017-04-28 at 9 18 54 pm" src="https://cloud.githubusercontent.com/assets/4672033/25552431/f613d33e-2c6f-11e7-8802-2a78b9708755.png">
<img width="333" alt="screen shot 2017-04-28 at 9 17 29 pm" src="https://cloud.githubusercontent.com/assets/4672033/25552430/f6103c9c-2c6f-11e7-9184-976a6e326550.png">
<img width="729" alt="screen shot 2017-04-28 at 9 16 31 pm" src="https://cloud.githubusercontent.com/assets/4672033/25552432/f6235c46-2c6f-11e7-912c-d3dd2cf1c200.png">
<img width="524" alt="screen shot 2017-04-28 at 8 12 52 pm" src="https://cloud.githubusercontent.com/assets/4672033/25552433/f6241c6c-2c6f-11e7-8a6f-234221c16579.png">
<img width="201" alt="screen shot 2017-04-28 at 8 11 37 pm" src="https://cloud.githubusercontent.com/assets/4672033/25552434/f6247d92-2c6f-11e7-877b-fe142d712e6f.png">
<img width="126" alt="screen shot 2017-04-28 at 8 07 48 pm" src="https://cloud.githubusercontent.com/assets/4672033/25552435/f62a0370-2c6f-11e7-8a47-a3fc415a2ace.png">
<img width="364" alt="screen shot 2017-04-28 at 7 53 09 pm" src="https://cloud.githubusercontent.com/assets/4672033/25552436/f62df14c-2c6f-11e7-9402-ab460f8644ea.png">

## Buttons changed as side effect

We had a global selector that changed every button that had `Button` className. I replaced that with `normalizeButton` so standard HTML5 buttons weren't affected. For reference:

<img width="95" alt="screen shot 2017-04-28 at 9 48 14 pm" src="https://cloud.githubusercontent.com/assets/4672033/25552426/f6073afc-2c6f-11e7-82cf-3c3187592698.png">
<img width="45" alt="screen shot 2017-04-29 at 12 17 23 am" src="https://cloud.githubusercontent.com/assets/4672033/25552495/9af421dc-2c71-11e7-9292-2baff9ccec02.png">
<img width="78" alt="screen shot 2017-04-29 at 12 17 07 am" src="https://cloud.githubusercontent.com/assets/4672033/25552498/9af7a780-2c71-11e7-9753-c4ce72674653.png">
<img width="64" alt="screen shot 2017-04-29 at 12 16 44 am" src="https://cloud.githubusercontent.com/assets/4672033/25552497/9af6a542-2c71-11e7-8cf9-6b7069657f8d.png">

luixxiul commented on app/renderer/components/common/browserButton.js:
```diff
@@ -0,0 +1,144 @@
+    }
+  },
+  // applies for primary and white buttons
+  browserButton_default: {
```
nice to see the modifed BEM style :-)

luixxiul commented:
Nice work. As a reminder, would you please apply the change to `about:styles`? Thanks!

cezaraugusto commented:
sure, thanks for checking ;)

cezaraugusto commented:
btw @luixxiul I added #8568 as an issue to track this PR since it doesn't refactor all components and removed #7168 reference (added you there as assignee as well) so we can keep tracking until every button uses BrowserButton. Then `<Button>` component can be removed. 

There are too many files using (now legacy) Button that the likeliness of regression was is high, so I decided to keep it gradual.

luixxiul commented:
FYI: reset Sync will be replaced with the primaryButton after #8044 is merged.

luixxiul commented:
@cezaraugusto I left some comments.

cezaraugusto commented:
thanks, updated

bbondy commented:
Since comments were addressed, merging.

luixxiul commented:
As I removed `.paymentsContainer` the margin between the buttons on "Back up your Brave wallet" dialog have regressed. The specification here seems to be no longer applied: https://github.com/brave/browser-laptop/blob/master/less/button.less#L180

![clipboard01](https://cloud.githubusercontent.com/assets/3362943/25771884/526e7426-3298-11e7-99c2-2824f0f825d5.jpg)

Since on the other dialogs the margin exists, I believe something is inconsistent between them. I'll tackle that.

srirambv commented:
@luixxiul is that expected behaviour for the backup wallet modal?

luixxiul commented:
No, and it was fixed already
